src/App.tsx

```diff
@@ -1,14 +1,23 @@
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import Tenants from "./pages/Tenants";
 import "./index.css";
+import Layout from "./pages/Layout";
+import { useState } from "react";
 
 function App() {
+  const [selectedView, setSelectedView] = useState("");
+
   return (
     <Router>
-      <Routes>
-        <Route path="/" element={<Tenants />} />
-        <Route path="/tenants" element={<Tenants />} />
-      </Routes>
+      <Layout setSelectedView={setSelectedView}>
+        <Routes>
+          <Route path="/" element={<Tenants selectedView={selectedView} />} />
+          <Route
+            path="/tenants"
+            element={<Tenants selectedView={selectedView} />}
+          />
+        </Routes>
+      </Layout>
```

Comment:
Please refer to this : https://medium.com/@dennisivy/creating-protected-routes-with-react-router-v6-2c4bbaf7bc1c